packages/core/src/types.ts

```diff
@@ -152,15 +153,17 @@ export function indexSearchParameterBundle(bundle: Bundle<SearchParameter>): voi
   }
 }
 
-/**
- * Indexes a SearchParameter resource for fast lookup.
- * Indexes by SearchParameter.code, which is the query string parameter name.
- * @param searchParam - The SearchParameter resource.
- * @see {@link IndexedStructureDefinition} for more details on indexed StructureDefinitions.
- */
-export function indexSearchParameter(searchParam: SearchParameter): void {
-  for (const resourceType of searchParam.base ?? []) {
-    let typeSchema = globalSchema.types[resourceType];
+export function indexDefaultSearchParameters(bundle: StructureDefinition[] | Bundle): void {
+  const sds = Array.isArray(bundle) ? bundle : (bundle.entry?.map((e) => e.resource as StructureDefinition) ?? []);
+  for (const sd of sds) {
+    if(sd.resourceType === 'StructureDefinition'){
+      getOrInitTypeSchema(sd.type);
+    }
+  }
+}
```

Comment:
It looks like the signature for this function is broader than it needs to be, and it could be simplified using one of our helper functions:
```suggestion
export function indexDefaultSearchParameters(bundle: Bundle): void {
  const sds = flatMapFilter(bundle.entry, (e) => e.resource?.resourceType === 'StructureDefinition' ? e.resource : undefined) ?? [];
  for (const sd of sds) {
    getOrInitTypeSchema(sd.type);
  }
}
```